src/lib/core/tt_hostname.h

```diff
@@ -0,0 +1,123 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <string.h>
+
+enum {
+	TT_HOSTNAME_LEN_MAX = 63,
+	TT_HOSTNAME_SIZE_MAX = TT_HOSTNAME_LEN_MAX + 1,
+};
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/**
+ * DNS- and host-friendly name. Only lowercase letters, digits, '-', starts with
+ * a letter, max 63 chars not including terminating zero.
+ */
```

Comment:
If you lean on some formal standard it worth to leave a link here.

Reply:
Thanks for the idea, done.

Reply:
Nice! Are we really want to forbid names like `queue123.example.org` and so on? It looks quite natural.

Reply:
Yes. That's a bad name for instance. Use `queue123` instead.

Reply:
It still looks quite natural, but now I know that you think that it is bad. Any reason behind this?

Reply:
Before a next redo of the patchset I need a resolution here. Either the comment needs to be resolved as is, or some new decision needs to be made.

Reply:
That's the answer, thanks!

My understanding is the following.

We don't see anything bad in DNS-like names ATM, but we forbid them just in case. Maybe to use in some other hierarchy like `<instance>.<replicaset>.<cluster>.<organization>` later.

I don't mind. Just want to know.

Closing the discussion.